code/modules/mob/mob_say.dm

```diff
@@ -80,7 +80,10 @@
 	log_talk(message, LOG_SAY, tag="DEAD")
 	if(SEND_SIGNAL(src, COMSIG_MOB_DEADSAY, message) & MOB_DEADSAY_SIGNAL_INTERCEPT)
 		return
-	deadchat_broadcast(rendered, source, follow_target = src, speaker_key = key)
+	var/displayed_key = key
+	if(src.client.holder?.fakekey)
+		displayed_key = null

```

Comment:
shouldn't this be
```
		displayed_key = client.holder.fakekey
```

Reply:
As of now deadchat messages shouldn't tie to a stealthmin's key at all to prevent exposing it.